site/src/components/WorkspaceActions/constants.ts

```diff
@@ -14,82 +14,77 @@ export enum ButtonTypesEnum {
   // disabled buttons
   canceling = "canceling",
   disabled = "disabled",
-  queued = "queued",
-  loading = "loading",
+  pending = "pending",
 }
 
 export type ButtonMapping = {
   [key in ButtonTypesEnum]: ReactNode
 }
 
-type StateActionsType = {
-  [key in WorkspaceStateEnum]: {
+type StateActionsType = Record<
+  WorkspaceStatus,
+  {
     primary: ButtonTypesEnum
     secondary: ButtonTypesEnum[]
     canCancel: boolean
   }
-}
+>
 
 // A mapping of workspace state to button type
 // 'Primary' actions are the main ctas
 // 'Secondary' actions are ctas housed within the popover
 export const WorkspaceStateActions: StateActionsType = {
-  [WorkspaceStateEnum.starting]: {
+  starting: {
```

Comment:
One of the reasons we made `WorkspaceStateEnum` was so that we could have a definitive mapping of the state returned from the BE to button UI state. Although this simplification makes my heart sing, I wonder if we're losing a little bit of that clarity and safety here. It is no longer immediately apparent that the keys in this data structure map to the types in the API's `WorkspaceStatus`. And the keys here aren't really typed anymore, right? This is a case where I really love enums and wish the API were returning an enum. It would make indexing much easier. @presleyp or @jsjoeio are there any easy workarounds with string literals I'm missing?

Reply:
I think it's still typesafe because the keys have to be values of `WorkspaceStatus`. I could be misremembering where I did this check, but I think it complained if I added a key that didn't belong or left one out.

Reply:
Ahhhhh, I didn't even see that. You're right - still typesafe! Feel free to ignore the clarify part of my comment - I'm just still figuring out where I land on this whole enum literal debate :)

Reply:
Me too honestly! Just following the generated type.

Reply:
Good point! I left a comment above related to this.